### ISSUE TITLE
Blog post detail page

### ISSUE THREAD

**Issue by kristinashu** [mozilla/foundation.mozilla.org#2911]
Based on - Blog design epic #2797 

We need to design a basic blog post detail page, let's start simple and then add complexity page. 

We will need to:
- [x] Benchmark other blogs
- [x] Write requirements list
- [x] Design and get feedback

Design:
- [x] responsive states
- [x] hover states

**Comment by kristinashu** [comment]
Gotta add medium as a benchmark! 

![medium](https://user-images.githubusercontent.com/1068783/55115616-265c0680-50a2-11e9-85dc-d71ad9d1182e.jpg)

**Comment by kristinashu** [comment]
@alanmoo a few tech questions before starting on the mvp design. Does wagtail already have some sort of pre-build blog template we can work off of? Can we use some of our existing components in the blog?

**Comment by alanmoo** [comment]
No built in template, just parts from other pages we can inherit.

We can use a streamfield as the body of the page with some/all of the available components as options, but my gut is telling me that we should stick with a rich text field to start. Though migrating from rich text to being able to use streamfield components down the line might get tricky. @Pomax do you have thoughts?

**Comment by kristinashu** [comment]
@beccaklam @alanmoo @xmatthewx [here's an initial list](https://docs.google.com/document/d/1O0EvKeEAo9yv7sgrlk-Rm2M7sPT8Y85esoKez8PZx94/edit?usp=sharing) of mvp vs long-term blog features, please add your thoughts!

**Comment by kristinashu** [comment]
Here are some _very_ quick comps https://redpen.io/p/bha12cf7387de4f7e7

**Comment by beccaklam** [comment]
Updated WIP comps: https://redpen.io/nzd574027ab66961ee @kristinashu

**Comment by kristinashu** [comment]
Note from convo with Alan: 
> Streamfield for body is probably best so that we can insert quote blocks, but let’s keep the page to a limited amount of streamfield component options (paragraph, quote, ….?)

So for the body content section of the blog, we will be able to use any of the components we already have on /opportunity type pages but we should limit the options. 

I've checked off the ones I think think we need but @beccaklam please update or add your thoughts:
- [x] Paragraph
- [x] Image
- [ ] Image text _(maybe?)_
- [ ] Image text mini _(maybe?)_
- [ ] Image grid _(maybe?)_
- [x] Video
- [ ] iFrame _(maybe?)_
- [x] Link button
- [x] Spacer
- [x] Quote
- [ ] Pulse listing
- [ ] Profile listing
- [ ] Profile by id
- [ ] Profile directory
- [ ] Airtable

<img width="637" alt="image" src="https://user-images.githubusercontent.com/1068783/55648145-630bba00-5794-11e9-9fde-56243b51fb0e.png">

**Comment by beccaklam** [comment]
Thanks @kristinashu I added "Image text mini" because I feel that staff often use blog posts to announce things that require a CTA. We could start with a smaller set of components but wondering if it will be hard to add more components if we find that staff keep asking for a particular thing? @alanmoo

**Comment by alanmoo** [comment]
Nope, easy to add.

**Comment by kristinashu** [comment]
I like that approach! Could you mock up a version with those selected components so we can show off what the blog will be able to do? Then Anil will share it around.

**Comment by beccaklam** [comment]
Also added "Image Text" since our current blog post is already using that component?

**Comment by beccaklam** [comment]
New blog detail page with mentioned components (I wasn't sure what our video looks like so I made it up for now): @kristinashu 
![blog-detail@2x](https://user-images.githubusercontent.com/20977376/55744117-1d542900-5a02-11e9-8e0a-6787d8ae62ae.jpg)

**Comment by beccaklam** [comment]
Updated MVP with large body copy: @kristinashu 
![blog-detail-01@2x](https://user-images.githubusercontent.com/20977376/55748371-77f28280-5a0c-11e9-8cce-f38eb72a6404.jpg)

**Comment by kristinashu** [comment]
Totally forgot about mobile... again! Should be fairly straight forward but can you please mock that up as well?

**Comment by beccaklam** [comment]
Oi, I should really start doing mobile first ...

**Comment by beccaklam** [comment]
Mobile mock ups have been added to the redpen: https://redpen.io/p/bha12cf7387de4f7e7

**Comment by beccaklam** [comment]
Added tablet design as well: https://redpen.io/wy8089751c4d20c936

**Comment by beccaklam** [comment]
MVP Requirements Checklist:
- [ ] Title field
- [ ] Author field
- [ ] Stream field includes paragraph component

Post MVP Requirements Checklist:
- [ ] Title field
- [ ] Author component, includes picture and link to Pulse profile (? @kristinashu is that where the link would go?)
- [ ] Stream field to include:
- Paragraph
- Image
- Image text
- Image text mini 
- Video
- Link button
- Spacer
- Quote
- [ ] Sticky Side Share buttons (need to ask devs if this is possible)
- [ ] Tags
- [ ] Share buttons after the blog post
- [ ] Related Content Section/Component, able to share up to three posts

**Comment by kristinashu** [comment]
Nice! A few thoughts:
- I think MVP could include all the selected stream fields since I assume it's similar amount of work to add one than it is to add the eight listed
- we won't be ready to add clickable tags and clickable author yet since that would require designing a new pages (I image clicking one would take a user to a page that shows all posts by that author or with that tag) (but might be nice to have the tag field in the cms so that when we do have the clickable functionality we won't have to go back and add tags in later)
- I think Share as a high priority than Related content (probably as separate implementation tickets)
- Maybe also add a note about how the sticky buttons are hidden on mobile
- And some notes about how the stream field components (except for paragraph) should follow the same style as existing ones

Just hover states and then I think you can open a couple implementation tickets!

**Comment by beccaklam** [comment]
Added hover states: https://redpen.io/an37406124d1e928f0

**Comment by beccaklam** [comment]
Will open implementation tickets tomorrow :)

**Comment by beccaklam** [comment]
Created implementation tickets #3029 (MVP) and #3030 (Post MVP) and closing this one!
